inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/context/ContextManagerImpl.java

```diff
@@ -20,24 +23,43 @@
      */
     public static final String BEAN_NAME = "contextManager";
 
+    private CommonTagsManager commonTags;
+
+    private InstrumentationConfigurationResolver configProvider;
+
     /**
      * Cache for storing the context objects.
      */
     private final Cache<Thread, Context> storedContexts = CacheBuilder.newBuilder().weakKeys().build();
 
+    public ContextManagerImpl(CommonTagsManager commonTags, InstrumentationConfigurationResolver configProvider) {
+        this.commonTags = commonTags;
+        this.configProvider = configProvider;
+    }
+
     @Override
     public Runnable wrap(Runnable r) {
-        return Context.current().wrap(r);
+        Runnable result;
+        try (val upPropagationBarrier = enterNewContext()) {
```

Comment:
I am still confused about why we need to open a new context every time we switch the threads.. I guess this is there to purely ensure that everything added by the user to the OC tag contact is combined with our latest context and propagated together. But in fact, both will be propagated in different GRPC keys, so this for me is a huge overhead.. Let's sync on this..